pywinauto/Linux/keyboard.py

```diff
@@ -0,0 +1,547 @@
+        if isinstance(self.key, str):
+            key = self.key
+            self.key = Xlib.XK.string_to_keysym(self.key)
+            if self.key == 0:
+                self.key = Xlib.XK.string_to_keysym(spec_keysyms[key])
+            self.key = _display.keysym_to_keycode(self.key)
+            if self.key == 0:
+                print("No button found")
+                return
+            self.is_shifted = key.isupper() or key in '~!@#$%^&*()_+{}|:"<>?'
+
+        self._key_modifiers(self.ctrl, (self.shift or self.is_shifted),
+                            self.alt, action = X.KeyPress)
+        if self.up:
+            fake_input(_display, X.KeyPress, self.key)
+            _display.sync()
+        if self.down:
+            fake_input(_display, X.KeyRelease, self.key)
+            _display.sync()
+        self._key_modifiers(self.ctrl, (self.shift or self.is_shifted),
+                            self.alt, action = X.KeyRelease)
+        _display.sync()
+
+    def _get_down_up_string(self):
+        """Return a string that will show whether the string is up or down
+
+        return 'down' if the key is a press only
+        return 'up' if the key is up only
+        return '' if the key is up & down (as default)
+        """
+        down_up = ""
+        if not (self.down and self.up):
+            if self.down:
+                down_up = "down"
+            elif self.up:
+                down_up = "up"
+        return down_up
+
+    def key_description(self):
+        """Return a description of the key"""
+        desc = "%s"% self.key
+        return desc
+
+    def __str__(self):
+        """Return key with modifiers as a string"""
+        parts = []
+        parts.append(self.key_description())
+        up_down = self._get_down_up_string()
+        if up_down:
+            parts.append(up_down)
+
+        return "<%s>"% (" ".join(parts))
+    __repr__ = __str__
+
+
+class PauseAction(KeyAction):
+
+    """Represents a pause action"""
+
+    def __init__(self, how_long):
+        self.how_long = how_long
+
+    def Run(self):
+        """Pause for the lenght of time specified"""
+        time.sleep(self.how_long)
+
+    def __str__(self):
+        return "<PAUSE %1.2f>"% (self.how_long)
+    __repr__ = __str__
+
+
+def handle_code(code):
+    """Handle a key or sequence of keys in braces"""
+    code_keys = []
+    # it is a known code (e.g. {DOWN}, {ENTER}, etc)
+    if code in CODES:
+        code_keys.append(KeyAction(CODES[code]))
+
+    # it is an escaped modifier e.g. {%}, {^}, {+}
+    elif len(code) == 1:
+        code_keys.append(KeyAction(code))
+
+    # it is a repetition or a pause  {DOWN 5}, {PAUSE 1.3}
+    elif ' ' in code:
+        to_repeat, count = code.rsplit(None, 1)
+        if to_repeat == "PAUSE":
+            try:
+                pause_time = float(count)
+            except ValueError:
+                raise KeySequenceError('invalid pause time %s'% count)
```

Comment:
Please use `.format()` instead of `%` string interpolation. Like so: `'invalid pause time {}'.format(count)`